apps/server/src/modules/task/service/task.service.spec.ts

```diff
@@ -218,4 +223,132 @@ describe('TaskService', () => {
+				DomainOperationReportBuilder.build(OperationType.DELETE, 1, [taskWithoutCourse.id]),
+				DomainOperationReportBuilder.build(OperationType.UPDATE, 2, [taskWithCourse.id, finishedTask.id]),
+			]);
+
+			return {
+				creator,
+				expectedResultByOnlyCreator,
+				expectedResultWithCreatorInTask,
+				expectedResultForFinishedTask,
+				expectedResult,
+			};
+		};
+
+		describe('when deleteUserData', () => {
+			it('should call deleteTasksByOnlyCreator with userId', async () => {
+				const { creator, expectedResultByOnlyCreator } = setup();
+				jest.spyOn(taskService, 'deleteTasksByOnlyCreator').mockResolvedValueOnce(expectedResultByOnlyCreator);
+
+				await taskService.deleteUserData(creator.id);
+
+				expect(taskService.deleteTasksByOnlyCreator).toHaveBeenCalledWith(creator.id);
+			});
+
+			it('should call removeCreatorIdFromTasks with userId', async () => {
+				const { creator, expectedResultWithCreatorInTask } = setup();
+				jest.spyOn(taskService, 'removeCreatorIdFromTasks').mockResolvedValueOnce(expectedResultWithCreatorInTask);
+
+				await taskService.deleteUserData(creator.id);
+
+				expect(taskService.removeCreatorIdFromTasks).toHaveBeenCalledWith(creator.id);
+			});
+
+			it('should call removeUserFromFinished with userId', async () => {
+				const { creator, expectedResultForFinishedTask } = setup();
+				jest.spyOn(taskService, 'removeUserFromFinished').mockResolvedValueOnce(expectedResultForFinishedTask);
+
+				await taskService.deleteUserData(creator.id);
+
+				expect(taskService.removeUserFromFinished).toHaveBeenCalledWith(creator.id);
+			});
+
+			it('should return domainOperation object with information about deleted user data', async () => {
+				const {
+					creator,
+					expectedResult,
+					expectedResultForFinishedTask,
+					expectedResultWithCreatorInTask,
+					expectedResultByOnlyCreator,
+				} = setup();
+
+				jest.spyOn(taskService, 'removeUserFromFinished').mockResolvedValueOnce(expectedResultForFinishedTask);
+				jest.spyOn(taskService, 'removeCreatorIdFromTasks').mockResolvedValueOnce(expectedResultWithCreatorInTask);
+				jest.spyOn(taskService, 'deleteTasksByOnlyCreator').mockResolvedValueOnce(expectedResultByOnlyCreator);
+
+				const result = await taskService.deleteUserData(creator.id);
+
+				expect(result).toEqual(expectedResult);
+			});
+		});
+	});
+
+	describe('handle', () => {
+		const setup = () => {
+			const targetRefId = new ObjectId().toHexString();
+			const targetRefDomain = DomainName.TASK;
+			const deletionRequest = deletionRequestFactory.build({ targetRefId, targetRefDomain });
+
+			const expectedData = DomainDeletionReportBuilder.build(DomainName.TASK, [
+				DomainOperationReportBuilder.build(OperationType.UPDATE, 2, [
+					new ObjectId().toHexString(),
+					new ObjectId().toHexString(),
+				]),
+			]);
+
+			return {
+				deletionRequest,
+				expectedData,
+			};
+		};
+
+		describe('when UserDeletedEvent', () => {
```

Comment:
Same here - please add a word or two to have any background on the test case's circumstances.

Reply:
done